app/controllers/imports_controller.rb

```diff
@@ -47,6 +48,24 @@ def load_csv
     end
   end
 
+  def upload_csv
+    raw_csv = params[:file].read
+
+    begin
+      csv = Import::Csv.new(raw_csv)
+      if csv.valid?
+        render turbo_stream: turbo_stream.action(
+          :update_input, "raw_csv_text_area", raw_csv
+        )
+      else
+        render json: { message: "CSV contents is not valid" }, status: :unprocessable_entity
+      end
+    rescue CSV::MalformedCSVError => error
+      file_extension = params[:file].path.split(".").last
+      render json: { message: "Expected file format CSV, but recieved #{file_extension}", error: error }, status: :bad_request
+    end
+  end
```

Comment:
This Turbo Stream solution definitely works, but I think we should try to build this flow without any dependencies on Streams to start as [outlined in the Turbo handbook](https://turbo.hotwired.dev/handbook/streams#progressively-enhance-when-necessary):

> It’s good practice to start your interaction design without Turbo Streams. Make the entire application work as it would if Turbo Streams were not available, then layer them on as a level-up. This means you won’t come to rely on the updates for flows that need to work in native applications or elsewhere without them.

I'm thinking we should build this endpoint a bit closer to what @tonyvince had in #973:

```rb
def upload_csv
  file = import_params[:raw_csv_str]
  @import.raw_csv_str = file.read
  CSV.parse(@import.raw_csv_str)
  if @import.save
    redirect_to configure_import_path(@import), notice: t(".import_loaded")
  else
    flash.now[:error] = @import.errors.full_messages.to_sentence
    render :load, status: :unprocessable_entity
  end
rescue CSV::MalformedCSVError, ArgumentError => error
  flash.now[:error] = error.message
  render :load, status: :unprocessable_entity
end
```

I know some of the designs dictate client-side interactions, but I'm 100% fine deviating a little bit from them for now so that we can achieve the simplest possible flow that's easy to test.